<a href="https://colab.research.google.com/github/Bestakdv/vhs-intro-ai/blob/main/Titanic_data_exploration.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In this project, you will explore the famous Titanic dataset to understand how data can provide insights into the survival of passengers. You will use the Pandas library in Python to analyze and visualize the data.

*Note: Points for each section are given in [].*


In [116]:
# Import libraries [6]
# numpy
# pandas
# matplotlib
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from google.colab import drive

# Get Data
Upload `train.csv` to a Google folder named `vhs-ai-intro`. Load this file into a dataframe named `df`. [6]





In [117]:
# access file and load into data frame
drive.mount('/content/drive')
#gets filepath of the csv file and stores it in variable
filepath = '/content/drive/MyDrive/train.csv'
#stores the read csv file into variable
df = pd.read_csv(filepath)

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


# Explore Data

In [118]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


Look at the output of `df.info()` above and answer the following: [8]

1. How many rows and columns does `df` have?
Df has 891 rows and 12 columns
2. Identify the columns that have integer datatype.
1st (#0), 2nd (#1), 3rd (#2), 7th (#6), and 8th(#7) columns
3. Identify the columns containing `null` (empty) values and calculate the proportion of values that are `null`.
column
The 6th column (#5), 11th column (#10), and 12th column (#11) have null values and the proportion of values that are null columns is 1/4






# Data Cleaning

In [119]:
df = df.drop(['PassengerId', 'Name', 'Ticket'], axis=1)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 9 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Survived  891 non-null    int64  
 1   Pclass    891 non-null    int64  
 2   Sex       891 non-null    object 
 3   Age       714 non-null    float64
 4   SibSp     891 non-null    int64  
 5   Parch     891 non-null    int64  
 6   Fare      891 non-null    float64
 7   Cabin     204 non-null    object 
 8   Embarked  889 non-null    object 
dtypes: float64(2), int64(4), object(3)
memory usage: 62.8+ KB


Explain what the above line of code is doing. You can use `df.info()` to check the changes in `df`. [2]

[ANSWER HERE]
The line of code changes the df by removing the PassengerId, Name and Ticket columns

In [120]:
df['Age'].fillna(df['Age'].median(), inplace=True)

Using the `fillna()` function, we can fill the missing values in the `Age` column with the median value of `Age`.

What does the `inplace` parameter specify in the `fillna()` function? [2]
The inplace parameter specifies if you want to modify the DataFrame directly or not.

# Data Analysis

Let's do some counting. Using `groupby` and `size()`, or otherwise:

1. Count the number of male and female passengers. (*Hint: Use `Sex` column*) [2]
2. What percentage of passengers survived? (*Hint: Use `Survived` column*) [2]


In [121]:
# Number of male & female passengers
passengers = df.groupby('Sex').size()
print(passengers)

Sex
female    314
male      577
dtype: int64


In [126]:
# Percentage of passengers that survived
survivalCount = df.loc[:, ['Survived']].groupby('Survived').size()
survivedPassengers = (survivalCount.loc[1] / len(df['Survived'])) * 100


print(survivedPassengers)

38.38383838383838


We want to see if gender played a role in the survival rates. Let us visualise the survival rates. We will combine the 2 columns `Survived` and `Sex` to answer this and use `Matplotlib` to plot the counts.

In [ ]:
# Group the data by 'Survived' and 'Sex' and get the counts
survival_gender_counts = df.groupby(['Survived', 'Sex']).size().unstack()

# Plot the grouped data
survival_gender_counts.plot(kind='bar')

# Add title and labels
plt.title('Survival by Gender')
plt.xlabel('Survived')
plt.ylabel('Count')

# Show the plot
plt.show()
print(type(survival_gender_counts))

Let us pay attention to the code.


1.   `survival_gender_counts` holds the counts of number of survivors that are male and female. What is the type of this variable? How many rows and columns does this variable have? [2]

[ANSWER HERE] The type of the variable is a DataFrame and it has 2 rows and 2 columns
2.   In the `survival_gender_counts.plot(kind='bar')` line, what does the `kind='bar'` parameter specify? How would the plot change if you used `kind='barh'` instead? [4]

[ANSWER HERE] kind=bar parameter specifies the type of graph, in this case you are specifying it to be a bar graph. If you used kind=barh instead then it would make a horizontal bar plot

We can create the same plot with another library `Seaborn` that is built on top of `Matplotlib`.

In [ ]:
import seaborn as sns
sns.countplot(x='Survived', hue='Sex', data=df)

plt.title('Survival by Gender')
plt.show()

We used the function `countplot()` in `seaborn` to generate the bar plot showing the counts we need.

Explain what each of the parameters for `countplot()` specify. [6]

[ANSWER HERE]
The first paramater gets the input data for X axis, the 2nd parameter visualizes the data of different categories, and the 3rd parameter get's the dataset for plotting

By looking at the plot, do you think gender and survival were related? Explain why or why not.    [3]

[ANSWER HERE]
Based on the plot I think gender and survival were related as mostly males died and approxamitly double amount of females survived than men. Also men died approximatly 4 times than women did.

In the Titanic dataset, `Pclass` refers to the passenger class, which indicates the socioeconomic status (SES) of the passengers. There are three classes:

* 1 = First class (Upper class)
* 2 = Second class (Middle class)
* 3 = Third class (Lower class)

Using `seaborn` library, create a count plot (similar to the one above) to show the relationship between survivors and passenger class.

In [ ]:
# Plot Survival by Passenger Class [6]
sns.countplot(x='Survived', hue='Pclass', data=df)

plt.title('Survival by Passenger Class')
plt.show()

By looking at the plot, do you think passenger class and survival were related? Explain why or why not. [3]

[ANSWER HERE]
I think the passenger class and survival were only slightly related as most of the people survived were in 1st class, however there was not that big of a difference between the classes.